### PR TITLE
 BREAKING: Move from styled-components to emotion

### PR DESCRIPTION
**Pull request by misund**
### Issues
- [x] emotion does not handle `.attrs()`. Refactor to [using recompose's `withProps()` instead](https://emotion.sh/docs/with-props)
- [x] emotion does not handle `.extend` calls. Refactor to `styled()` calls
- [x] emotion has deprecated using props interpolations in css`` calls. See deprecation warning in the console for workaround

#### Please tick a box ###
- [x] **refactor** _(A code change that neither fixes a bug nor adds a feature_)
- [x] **performance** _(A code change that improves performance_)

#### If you're adding a feature, is it documented in a storybook story?
- [x] Not adding a new feature

#### Are the components you're working on reusable between brands?
- [x] Yes _(Using variables that are defined in the default theme)_

#### If you're creating markup, did you add proper semantics? 
- [x] Not applicable

_(Did you do a CR and see that there is something that we should check for each PR, that are not on the list, please [update this document](https://github.com/dbmedialab/shiny/edit/master/.github/PULL_REQUEST_TEMPLATE.md))_

### CODE REVIEW

**Comment by sergeyvlakh** [comment]
Me waiting for percy: :popcorn: :popcorn: :popcorn:

**Comment by misund** [comment]
I'm expecting the grid to deviate a bit. Switched libraries to react-emotion-flexboxgrid for the moment, so we won't have to wait for LoicMahieu/react-styled-flexboxgrid#146.

However, react-emotion-flexboxgrid was forked out before LoicMahieu/react-styled-flexboxgrid#87, so I think I might have to reintroduce our workaround from #10b9107960146ee

**Comment by misund** [comment]
> No stories were found.

**Comment by misund** [comment]
I'll add issues in a checklist in the description as I discover them

**Comment by misund** [comment]
> 86 unreviewed changes
> in 86 total snapshots

Could be worse 😂

I'm gone for vacation for about a week, but I think this actually is pretty close.

**Comment by SindreSvendby** [comment]
Percy Build: https://percy.io/dbmedialab/shiny/builds/963812 had only 10 diffs wrong
That is git hash: https://github.com/dbmedialab/shiny/commit/2832ed66c3bcd719bdbb252a835ecad5436e2606

File: src/atoms/BorderedButton.js [hunk 1]
```diff
@@ -27,19 +26,11 @@ const BorderedButton = styled(Button)`
...
 	${props => props.isActive && coloredBackground};
-
 	@media (min-width: ${props => props.theme.flexboxgrid.breakpoints.md}em) {
 		&:hover{
 			${coloredBackground}
 		};
 	}
 `;
 
-BorderedButton.defaultProps = {
```
Comment (SindreSvendby) [inline src/atoms/BorderedButton.js]: why is this removed?

File: src/atoms/BylineWithTwoLines.js [hunk 2]
```diff
@@ -1,69 +1,78 @@
...
-    width: 4.5rem;
-    height: 4.5rem;
-    overflow: hidden;
-    border-radius: 50%;
-    object-fit: cover;
-    margin-right: ${getVariable('horizontalBase')};
+const BylineBlock = styled.div``;
+
+
+const BylineImage = (props) => {
```
Comment (SindreSvendby) [inline src/atoms/BylineWithTwoLines.js]: Could be a better way, but this way worked :) 

It was implemented with `recompose/withComponent`  but that implementation did not work

File: src/atoms/BylineWithTwoLines.js [hunk 3]
```diff
@@ -1,69 +1,78 @@
...
+};
+
+const StyledBylineImage = styled(BylineImage)`
+	float: left;
+	width: 4.5rem;
+	height: 4.5rem;
+	overflow: hidden;
+	border-radius: 50%;
+	object-fit: cover;
+	margin-right: ${props => getVariable('horizontalBase')(props)};
```
Comment (SindreSvendby) [inline src/atoms/BylineWithTwoLines.js]: `${props ... (props)}` syntax not need

File: src/atoms/BylineWithTwoLines.js [hunk 4]
```diff
@@ -1,69 +1,78 @@
...
-    height: 4.5rem;
-    overflow: hidden;
-    border-radius: 50%;
-    margin-right: ${getVariable('horizontalBase')};
-    text-align: center;
+	width: 4.5rem;
+	height: 4.5rem;
+	overflow: hidden;
+	border-radius: 50%;
+	margin-right: ${props => getVariable('horizontalBase')(props)};
```
Comment (SindreSvendby) [inline src/atoms/BylineWithTwoLines.js]: `${props ... (props)}` syntax not need

File: src/atoms/BylineWithTwoLines.js [hunk 5]
```diff
@@ -1,69 +1,78 @@
...
+	text-align: center;
 `;
 
-const TextBlock = Styled.div`
-    float: left;
-    color: ${getColor('type')};
-    font-family: ${getVariable('mainFont')};
+const TextBlock = styled.div`
+	float: left;
+	color: ${props => getColor('type')(props)};
```
Comment (SindreSvendby) [inline src/atoms/BylineWithTwoLines.js]: `${props ... (props)}` syntax not need

File: src/atoms/BylineWithTwoLines.js [hunk 6]
```diff
@@ -1,69 +1,78 @@
...
+	font-family: ${props => getVariable('mainFont')(props)};
```
Comment (SindreSvendby) [inline src/atoms/BylineWithTwoLines.js]: `${props ... (props)}` syntax not need

File: src/atoms/BylineWithTwoLines.js [hunk 7]
```diff
@@ -1,69 +1,78 @@
...
 `;
 
-const Name = Styled.div`
-    font-size: ${getVariable('uiRegularSize')};
-    line-height: ${getVariable('uiRegularLineHeight')};
-    font-weight: 300;
+const Name = styled.div`
+	font-size: ${props => getVariable('uiRegularSize')(props)};
```
Comment (SindreSvendby) [inline src/atoms/BylineWithTwoLines.js]: `${props ... (props)}` syntax not need

File: src/atoms/BylineWithTwoLines.js [hunk 8]
```diff
@@ -1,69 +1,78 @@
...
+	line-height: ${props => getVariable('uiRegularLineHeight')(props)};
```
Comment (SindreSvendby) [inline src/atoms/BylineWithTwoLines.js]: `${props ... (props)}` syntax not need

File: src/atoms/BylineWithTwoLines.js [hunk 9]
```diff
@@ -1,69 +1,78 @@
...
+	font-weight: 300;
 `;
 
-const Email = Styled.a.attrs({
-	href: ({ email }) => `mailto:${email}`,
-})`
-    font-size: ${getVariable('uiSmallSize')};
-    font-weight: 300;
-    color: ${getColor('type')};
+const Email = ({ email, children }) => <a href={`mailto:${email}`}>{children}</a>;
```
Comment (SindreSvendby) [inline src/atoms/BylineWithTwoLines.js]: Same comment as byline image

File: src/atoms/DymoHeading.js [hunk 10]
```diff
@@ -12,52 +12,66 @@ const Hgroup = styled.hgroup`
...
-	a:hover & {
-		color: ${props => getColor(props.skin.textColor, props.skin.textHoverShade)};
-		background: ${props => getColor(props.skin.backgroundColor, props.skin.backgroundHoverShade)};
-		box-shadow:
-			calc(1/2 * ${getVariable('horizontalBase')})
-			0
-			${props => getColor(props.skin.backgroundColor, props.skin.backgroundHoverShade)}
-		;
-	}
+	${(props) => {
```
Comment (SindreSvendby) [inline src/atoms/DymoHeading.js]: This looks good <3

File: src/atoms/Grid.js [hunk 11]
```diff
@@ -1 +1,27 @@
...
+
+const Grid = styled.div`
+  margin-right: auto;
+  margin-left: auto;
+  padding-right: ${p => `${config(p).outerMargin}rem`};
+  padding-left: ${p => `${config(p).outerMargin}rem`};
+  ${p => !p.fluid && css`
+  ${DIMENSION_NAMES.map((t) => {
+		return config(p).container[t] && config(p).media[t]`
+       width: ${config(p).container[t]}rem;
```
Comment (SindreSvendby) [inline src/atoms/Grid.js]: `Col`, `Row`, `Grid` and `grid-config`, is all taken from `react-styled-flexboxgrid` - almost no changes, could be merged upstream!  

I think the related changes is inside this map function that start on line 12

File: src/atoms/FrontHeading.js [hunk 12]
```diff
@@ -19,31 +19,31 @@ const FheadStyle = props => css`
...
+		border-bottom: .2rem solid ${getColor(props.lineColor)(props)};
 		vertical-align: middle;
     }
 `;
 
 const FrontSmallHeading = styled(SmallHeading)`
 	${FheadStyle}
-	@media screen and (min-width: ${getVariable('largeWidth')}) {
-		font-size: ${getVariable('headingSmallSize')};
+	@media screen and (min-width: ${props => getVariable('largeWidth')(props)}) {
```
Comment (SindreSvendby) [inline src/atoms/FrontHeading.js]: `${props ... (props)}` syntax not need

File: src/atoms/FrontHeading.js [hunk 13]
```diff
@@ -19,31 +19,31 @@ const FheadStyle = props => css`
...
+		font-size: ${props => getVariable('headingSmallSize')(props)};
```
Comment (SindreSvendby) [inline src/atoms/FrontHeading.js]: `${props ... (props)}` syntax not need

File: src/atoms/FrontHeading.js [hunk 14]
```diff
@@ -19,31 +19,31 @@ const FheadStyle = props => css`
...
 	}
 `;
 
 const FrontMediumHeading = styled(MediumHeading)`
 	${FheadStyle}
-	@media screen and (min-width: ${getVariable('largeWidth')}) {
-		font-size: ${getVariable('headingMediumSize')};
+	@media screen and (min-width: ${props => getVariable('largeWidth')(props)}) {
```
Comment (SindreSvendby) [inline src/atoms/FrontHeading.js]: `${props ... (props)}` syntax not need

File: src/atoms/FrontHeading.js [hunk 15]
```diff
@@ -19,31 +19,31 @@ const FheadStyle = props => css`
...
+		font-size: ${props => getVariable('headingMediumSize')(props)};
```
Comment (SindreSvendby) [inline src/atoms/FrontHeading.js]: `${props ... (props)}` syntax not need

File: src/atoms/FrontHeading.js [hunk 16]
```diff
@@ -19,31 +19,31 @@ const FheadStyle = props => css`
...
 	}
 `;
 const FrontLargeHeading = styled(LargeHeading)`
 	${FheadStyle}
-	font-size: ${getVariable('headingMediumSize')};
-	@media screen and (min-width: ${getVariable('largeWidth')}) {
-		font-size: ${getVariable('headingLargeSize')};
+	font-size: ${props => getVariable('headingMediumSize')(props)};
```
Comment (SindreSvendby) [inline src/atoms/FrontHeading.js]: `${props ... (props)}` syntax not need

File: src/atoms/FrontHeading.js [hunk 17]
```diff
@@ -19,31 +19,31 @@ const FheadStyle = props => css`
...
+	@media screen and (min-width: ${props => getVariable('largeWidth')(props)}) {
```
Comment (SindreSvendby) [inline src/atoms/FrontHeading.js]: `${props ... (props)}` syntax not need

File: src/atoms/FrontHeading.js [hunk 18]
```diff
@@ -19,31 +19,31 @@ const FheadStyle = props => css`
...
+		font-size: ${props => getVariable('headingLargeSize')(props)};
```
Comment (SindreSvendby) [inline src/atoms/FrontHeading.js]: `${props ... (props)}` syntax not need

File: src/atoms/Heading.js [hunk 19]
```diff
@@ -1,95 +1,93 @@
...
-	font-family: ${getVariable('headingsFont')};
-
-	${props => props.ALLCAPS && css`
-		text-transform: uppercase;
-		letter-spacing: .1rem;
-	`}
-
-	a {
-		color: inherit;
+	${(props) => {
```
Comment (SindreSvendby) [inline src/atoms/Heading.js]: :heart:

File: src/atoms/LinkBarLinkBase.js [hunk 20]
```diff
@@ -23,29 +23,29 @@ export const LinkBarLinkBase = styled.a`
...
+				calc(1/2 * (3/2*${getVariable('verticalBase')(props)} - ${getVariable('uiRegularLineHeight')(props)}) )
+				${getVariable('horizontalBase')(props)};
 		`;
 	}}
 
 	border: 0;
 	outline: none;
 	text-decoration: none;
-	font-family: ${getVariable('headingsFont')};
+	font-family: ${props => getVariable('headingsFont')(props)};
```
Comment (SindreSvendby) [inline src/atoms/LinkBarLinkBase.js]: `${props ... (props)}` syntax not need

File: src/atoms/LinkBarLinkBase.js [hunk 21]
```diff
@@ -23,29 +23,29 @@ export const LinkBarLinkBase = styled.a`
...
 	font-size: ${getVariable('uiRegularSize')};
-	line-height: ${props => (props.isBlockLink ? '0' : getVariable('uiRegularLineHeight'))};
+	line-height: ${props => (props.isBlockLink ? '0' : getVariable('uiRegularLineHeight')(props))};
```
Comment (SindreSvendby) [inline src/atoms/LinkBarLinkBase.js]: `${props ... (props)}` syntax not need 
(but, (props) need to be there)

File: src/atoms/Tag.js [hunk 22]
```diff
@@ -16,14 +17,14 @@ const TagSpanItem = styled.span`
 	text-transform: uppercase;
 	color: ${getColor('grey')};
 `;
-const TagAItem = TagSpanItem.withComponent('a').extend.attrs({
+const TagAItem = withProps({
```
Comment (SindreSvendby) [inline src/atoms/Tag.js]: I do not think this work, that could be way Shiny had a visual diff!
Need to check that up!

File: src/molecules/LunchButton.js [hunk 23]
```diff
@@ -12,9 +12,15 @@ import { BlockLinkWithBackground as Link } from '../atoms/BlockLinkWithBackgroun
 import { LunchKicker as Kicker } from '../atoms/LunchKicker';
 import { MediumHeading } from '../atoms/Heading';
 
+const styles = (props) => {
```
Comment (SindreSvendby) [inline src/molecules/LunchButton.js]: Could be inlined

File: src/utils/get-color.js [hunk 24]
```diff
@@ -30,3 +30,7 @@ export const getColor = (color, shade=null) => (props) => {
 
 	return props.theme.colors[shadedColorString];
 };
+
+const createCssRuleWithColor = (cssProp, cssValue) => css` ${cssProp}: cssValue`;
+
+export const createColorCss = (color, cssProp, shade=null) => createCssRuleWithColor(cssProp, getColor(color, shade));
```
Comment (SindreSvendby) [inline src/utils/get-color.js]: I think this is unused, check up and remove!

File: stories/typography/dice.js [hunk 25]
```diff
@@ -83,7 +85,7 @@ import { Dice } from '@aller/shiny';
 						</tr>
```
Comment (SindreSvendby) [inline stories/typography/dice.js]: Whats the deal with this syntax from github? Check up that it looks ok in editor!

Comment (SindreSvendby) [reply stories/typography/dice.js]: Looks ok in editor, just a github wtf
